tests/providers/aws/services/ec2/ec2_securitygroup_allow_ingress_from_internet_to_any_port/ec2_securitygroup_allow_ingress_from_internet_to_any_port_test.py

```diff
@@ -359,3 +359,144 @@ def test_ec2_default_sgs_with_all_ports_check(self):
+                security_group_id,
+                region,
+            )
+
+            # Ensure that is_failed_check returns True
+            assert EC2.is_failed_check(
+                ec2_securitygroup_allow_ingress_from_internet_to_any_port.__name__,
+                security_group_id,
+                region,
+            )
+
+    @mock_aws
+    def test_ec2_non_compliant_default_sg_pass_to_avoid_fail_twice(self):
+        # Create EC2 Mocked Resources
+        ec2_client = client("ec2", region_name=AWS_REGION_US_EAST_1)
+        ec2_client.create_vpc(CidrBlock="10.0.0.0/16")
+        default_sg = ec2_client.describe_security_groups(GroupNames=["default"])[
+            "SecurityGroups"
+        ][0]
+        default_sg_id = default_sg["GroupId"]
+        ec2_client.authorize_security_group_ingress(
+            GroupId=default_sg_id,
+            IpPermissions=[
+                {
+                    "IpProtocol": "-1",
+                    "FromPort": -1,
+                    "ToPort": -1,
+                    "IpRanges": [{"CidrIp": "0.0.0.0/0"}],
+                },
+                {
+                    "IpProtocol": "tcp",
+                    "FromPort": 22,
+                    "ToPort": 22,
+                    "IpRanges": [{"CidrIp": "0.0.0.0/0"}],
+                },
+            ],
+        )
+
+        from prowler.providers.aws.services.ec2.ec2_service import EC2
+
+        aws_provider = set_mocked_aws_provider(
+            [AWS_REGION_EU_WEST_1, AWS_REGION_US_EAST_1]
+        )
+
+        with mock.patch(
+            "prowler.providers.common.common.get_global_provider",
+            return_value=aws_provider,
+        ), mock.patch(
+            "prowler.providers.aws.services.ec2.ec2_securitygroup_allow_ingress_from_internet_to_all_ports.ec2_securitygroup_allow_ingress_from_internet_to_all_ports.ec2_client",
+            new=EC2(aws_provider),
+        ) as ec2_client_instance, mock.patch(
+            "prowler.providers.aws.services.ec2.ec2_securitygroup_allow_ingress_from_internet_to_all_ports.ec2_securitygroup_allow_ingress_from_internet_to_all_ports.vpc_client",
+            new=VPC(aws_provider),
+        ) as vpc_client_instance:
+            # Run check for all ports
+            from prowler.providers.aws.services.ec2.ec2_securitygroup_allow_ingress_from_internet_to_all_ports.ec2_securitygroup_allow_ingress_from_internet_to_all_ports import (
+                ec2_securitygroup_allow_ingress_from_internet_to_all_ports,
+            )
+
+            check_all_ports = (
+                ec2_securitygroup_allow_ingress_from_internet_to_all_ports()
+            )
+            result_all_ports = check_all_ports.execute()
+
+            # Verify that the all ports check has detected the issue
+            assert any(
+                sg.status == "FAIL" and sg.resource_id == default_sg_id
+                for sg in result_all_ports
+            )
```

Comment:
Maybe is easier just to call `AWSService.set_failed_check` and then run the `ec2_securitygroup_allow_ingress_from_internet_to_any_port` and verify that the finding's status extended is the new one.

Reply:
This comment applies to all the tests.